### ISSUE TITLE
Running NUFFT in parallel causes a seg fault in CPU kernel

### ISSUE THREAD

**Issue by jmontalt** [mrphys/tensorflow-nufft#7]
The following code reproduces the problem:

```
with tf.device('/cpu:0'):
  rank = 2
  num_points = 20000
  grid_shape = [128] * rank
  batch_size = 100
  rng = tf.random.Generator.from_seed(10)
  points = rng.uniform([batch_size, num_points, rank], minval=-np.pi, maxval=np.pi)
  source = tf.complex(tf.ones([batch_size, num_points]),
                      tf.zeros([batch_size, num_points]))
  @tf.function
  def parallel_nufft_adjoint(source, points):
    def nufft_adjoint(inputs):
      src, pts = inputs
      return nufft_ops.nufft(src, pts, grid_shape=grid_shape,
                             transform_type='type_1',
                             fft_direction='backward')
    return tf.map_fn(nufft_adjoint, [source, points],
                     parallel_iterations=4,
                     fn_output_signature=tf.TensorSpec(grid_shape, tf.complex64))
  
  result = parallel_nufft_adjoint(source, points)
```

**Comment by jmontalt** [comment]
This issue has been tracked down and will be addressed in release 0.5.0.

**Comment by jmontalt** [comment]
This issue has been fixed in release 0.5.0.
